guides/nodejs.md

```diff
@@ -0,0 +1,308 @@
+# Node.js Developer's Guide
+
+When creating a Node.js function using the `func` binary, the project directory
+looks like a typical Node.js project. Both HTTP and Event functions have the same
+template structure.
+
+```
+❯ tree
+.
+├── func.yaml
+├── index.js
+├── package.json
+├── README.md
+└── test
+    ├── integration.js
+    └── unit.js
+```
+
+Aside from the `func.yaml` file, this looks like the beginning of just about
+any Node.js project. For now, we will ignore the `func.yaml` file, and just
+say that it is a configuration file that is used when building your project.
+If you're really interested, check out the [reference doc](config-reference.doc).
```

Comment:
`[reference doc](config-reference.doc)` - that doc doesn't existing yet, but I guess that's ok for now.